packages/code-du-travail-frontend/src/common/Answer.js

```diff
@@ -45,97 +47,72 @@ function Answer({
   wide,
   footer,
   date,
-  icon,
   sourceType,
   additionalContent,
   breadcrumbs = [],
+  relatedItems = [],
   emptyMessage = "Aucun résultat"
 }) {
-  const [portalComponents, setPortalComponents] = useState();
-  useEffect(() => {
-    const nodes = Array.from(
-      document.querySelectorAll(
-        "[data-main-content] p, [data-main-content] li:not([role=tab])"
-      )
-    ).reduce((state, node) => {
-      glossary.forEach(item => {
-        // we cannot use \b word boundary since \w does not match diacritics
-        // So we do a kind of \b equivalent.
-        // the main différence is that matched pattern can include a whitespace as first char
-        const frDiacritics = "àâäçéèêëïîôöùûüÿœæÀÂÄÇÉÈÊËÎÏÔÖÙÛÜŸŒÆ";
-        const wordBoundaryStart = `(?:^|[^\\w${frDiacritics}])`;
-        const wordBoundaryEnd = `(?![\\w${frDiacritics}])`;
-        const patterns = [...new Set([item.title, ...item.variants])]
-          .map(
-            term =>
-              new RegExp(`${wordBoundaryStart}${term}${wordBoundaryEnd}`, "gi")
-          )
-          .concat(item.abbrs.map(abbr => new RegExp(`\\b${abbr}\\b`, "g")));
-
-        patterns.forEach(pattern => {
-          node.innerHTML = node.innerHTML.replace(pattern, function(match) {
-            if (new RegExp(`^[^\\w${frDiacritics}]`).test(match)) {
-              // Since match string can start with a space, we trim it and insert the space before the tooltip markup
-              return `${match.slice(0, 1)}<span data-tooltip-slug="${
-                item.slug
-              }" data-tooltip-term="${match.slice(1)}"></span>`;
-            }
-            return `<span data-tooltip-slug="${item.slug}" data-tooltip-term="${match}"></span>`;
-          });
-        });
-      });
+  const portalComponents = usePortals(children, html);
+  const linkedResults = groupByDisplayCategory(relatedItems);
```

Comment:
Since we are only interested in a subset of the results, why not filter the list befor on the api side ?

Reply:
In fact, we will probably use other methods than `getSearch()` to retrieve relatedTheme or relatedTools that's why I think it should be done on the API side

Reply:
It's better if it can be done on API Side!